addons/im_livechat/static/src/legacy/public_livechat_chatbot.js

```diff
@@ -0,0 +1,874 @@
+     * in this case).
+     *
+     * @private
+     */
+    _onChatbotRestartScript: async function (ev) {
+        this._chatWindow.$('.o_composer_text_field').removeClass('d-none');
+        this._chatWindow.$('.o_livechat_chatbot_end').addClass('d-none');
+
+        if (this.nextStepTimeout) {
+            clearTimeout(this.nextStepTimeout);
+        }
+
+        if (this.welcomeMessageTimeout) {
+            clearTimeout(this.welcomeMessageTimeout);
+        }
+
+        const postedMessage = await session.rpc('/im_livechat/chatbot_restart', {
+            channel_uuid: this._livechat.getUUID(),
+            chatbot_script_id: this._chatbot.chatbot_script_id
+        });
+
+        if (postedMessage) {
+            this._chatbotAddMessage(postedMessage);
+        }
+
+        this._chatbotCurrentStep = null;
+        this._chatbotSetIsTyping();
+        this.nextStepTimeout = setTimeout(
+            this._chatbotTriggerNextStep.bind(this), this._chatbotMessageDelay);
+    },
+
+    _onChatbotInputKeyDown: function () {
+        if (this._chatbotCurrentStep &&
+            this._chatbotCurrentStep.chatbot_step_type === 'free_input_multi') {
+            this._debouncedChatbotAwaitUserInput();
+        }
+    },
+
+    /**
+     * We need to save the selected answer right away to be safe in case we redirect to another page
+     * Indeed, on the other page, we need to know that an answer was selected to avoid showing
+     * the options again when we re-open the conversation with the bot.
+     *
+     * @param {MouseEvent} ev
+     * @private
+     */
+    _onChatbotOptionClicked: async function (ev) {
+        ev.stopPropagation();
+
+        const $target = $(ev.currentTarget);
+        const stepId = $target.closest('ul').data('chatbotStepId');
+        const selectedAnswer = $target.data('chatbotStepAnswerId');
+
+        const redirectLink = $target.data('chatbotStepRedirectLink');
+        if (redirectLink) {
+            this._isChatbotRedirecting = true;
+        }
+
+        await this._sendMessage({
+            content: $target.text().trim()
+        });
+
+        let messageId = false;
+        this._messages.forEach((message) => {
+            if (message.getChatbotStepId() === stepId) {
+                messageId = message.getID();
+                message.setChatbotStepAnswerId(selectedAnswer);
+                this._chatbotCurrentStep['chatbot_selected_answer_id'] = selectedAnswer;
+                this._renderMessages();
+                this._chatbotSaveSession();
+            }
+        });
+
+        const saveAnswerPromise = session.rpc('/im_livechat/chatbot_save_answer', {
+            channel_uuid: this._livechat.getUUID(),
+            message_id: messageId,
+            selected_answer_id: selectedAnswer
+        });
+
+        if (redirectLink) {
+            await saveAnswerPromise;  // ensure answer is saved before redirecting
```

Comment:
We will wait the response only if we need `redirectLink` (but we should always wait it imo)

Reply:
In other cases there is not really a point since the caller is a handler, which does not recognize the handler method as async anyway (AFAIK) and will not await its completion to do "something else".

I like specifying it that way because it marks it as extra important to "wait before redirecting".
Maybe I missed something, in that case feel free to enlighten me :smile: .

Reply:
> which does not recognize the handler method as async anyway (AFAIK)

Ok to ignore it then :)